src/sentry/models/useremail.py

```diff
@@ -21,7 +21,7 @@ class UserEmail(Model):
     __core__ = True
 
     user = FlexibleForeignKey(settings.AUTH_USER_MODEL, related_name="emails")
-    email = models.EmailField(_("email address"), max_length=75)
+    email = CIEmailField(_("email address"), max_length=75)
```

Comment:
Do we know how expensive this alter is? I'm not sure off hand what kind of locks are needed or if this is even safe to do in place without losing data.

Reply:
@mattrobenolt When you ask about how expensive is the alter, do you mean what's the penalty for locking the table while the migration is being run? Honestly, we might be able to away with the lock since it's incredibly unlikely that an insert would happen that would violate the new constraint. It's only happened 7 times this year in SaaS.

Reply:
I think you're misunderstanding what the lock is for. It'll have to rewrite the whole table potentially so the entire table is locked while that happens. I don' tknow if that's true for the transformation here from varchar -> citext. Hence why I'm asking if we know or not.

In worst case scenario, an exclusive lock will be needed, and it could cascade into taking down Sentry while it queues up behind SELECTS and block future SELECTS until it's able to acquire the lock needed. So it just depends.

Reply:
@mattrobenolt @wedamija I did a test on my machine to see how long it would take. I generated 600k items in the `UserEmail` table and then ran the now split up migrations. The alter table migration took 17 seconds on my machine. Does that help?

Reply:
@mattrobenolt I updated the PR description with a list of places where this table is used

Reply:
What is the SQL that the new index replacement generates? That would be helpful in figuring out which locks postgres will take.

Reply:
@markstory Here's what I see:
```
CREATE UNIQUE INDEX sentry_useremail_user_id_469ffbb142507df2_uniq ON sentry_useremail(user_id int4_ops,email citext_ops);
```

Reply:
I ran this locally:
```
sentry=# BEGIN TRANSACTION;
BEGIN
sentry=# alter table sentry_useremail ALTER COLUMN email type citext;
sentry=# select nspname,relname,l.* from pg_locks l join pg_class c on
 (relation=c.oid) join pg_namespace nsp on (c.relnamespace=nsp.oid) ;
```

There were definitely exclusive access locks taken out on the entire table. Not sure if this is the best way to determine lock usage though.

Reply:
That index creation SQL needs `CONCURRENTLY` to reduce the impact of locks taken.